roles/compute_post/tasks/main.yml

```diff
@@ -8,18 +23,18 @@
   notify:
     - Restart libvirt
 
+- name: make sure nova permissions are correct
+  file: path=/var/lib/nova owner=nova group=nova recurse=yes
+
 - meta: flush_handlers
 
 - name: Enable messagebus service
   service: name=messagebus state=started enabled=true
-  when: ansible_distribution == 'CentOS' and ansible_distribution_major_version == '6'
+  when: ansible_distribution == 'CentOS' and ansible_distribution_version.split(".")[0] == '6'
 
 - name: Upgrade iproute
   yum: name=iproute state=latest
-  when: ansible_distribution == 'CentOS' and ansible_distribution_major_version == '6'
-
-- name: Change nova dir permissions
-  file: path=/var/lib/nova state=directory owner='nova' group='nova' recurse=yes
```

Comment:
We are already doing that at line 26.
https://github.com/midonet/ansible-midonet/pull/2/files#diff-1af5c60a984b8d59440e3d919aa033baR26